packages/schematics/migrations/update-7-2-0/update-7-2-0.ts

```diff
@@ -0,0 +1,142 @@
+import {
+  Rule,
+  chain,
+  noop,
+  SchematicContext,
+  Tree
+} from '@angular-devkit/schematics';
+import { normalize, join, Path, dirname } from '@angular-devkit/core';
+
+import { relative } from 'path';
+
+import { updateJsonInTree, readJsonInTree } from '../../src/utils/ast-utils';
+import { getWorkspacePath } from '../../src/utils/cli-config-utils';
+import { offsetFromRoot } from '../../src/utils/common';
+import { stripIndents } from '@angular-devkit/core/src/utils/literals';
+
+function getBuilders(project: any): string[] {
+  return Array.from(
+    new Set(Object.values<any>(project.architect).map(target => target.builder))
+  );
+}
+
+const builderTypes: { [key: string]: string[] } = {
+  '@angular-devkit/build-angular:karma': ['jasmine'],
+  '@angular-devkit/build-angular:protractor': ['jasmine', 'jasminewd2'],
+  '@nrwl/builders:jest': ['jest', 'node'],
+  '@nrwl/builers:cypress': ['cypress']
+};
+
+function getTypes(host: Tree, project: any) {
+  let types = [];
+  types = types.concat(
+    ...getTsConfigs(project).map(tsconfigPath => {
+      const tsconfig = readJsonInTree(host, tsconfigPath);
+      return tsconfig.compilerOptions.types || [];
```

Comment:
I think the right thing to do is to set the whole thing to undefined if any of the files have it set to undefined.